src/lane_manager.cpp

```diff
@@ -68,89 +66,31 @@ bool LaneManager::intersection_collision_check() {
 		for (int j = i + 1; j < cars_in_intersec.size(); j++) {
 			Car* second_car = cars_in_intersec[j];
 			rect_bounding_box second_bb = second_car->get_bounding_box();
-			if (first_car->check_collision(second_bb.bottom_left)) {
-				collision_occurring = true;
-				int collided_triangle = mesh_collision_check(second_car, first_car, second_bb.bottom_left);
-				if (collided_triangle != -1) {
-					first_car->collided(collided_triangle);
-					second_car->collided(-1);
-					printf("first car getting hit by second bottom left\n");
-					//first_car->change_color();
-				}
-				//DEBUG
-
 
-				//TODO: If mesh_collision_check != -1, apply collision depending on triangle returned
-			}
-			else if (first_car->check_collision(second_bb.bottom_right)) {
+			// First car is victim
+			if (first_car->check_collision(second_bb.bottom_left)
+				|| first_car->check_collision(second_bb.bottom_right)
+				|| first_car->check_collision(second_bb.top_right)
+				|| first_car->check_collision(second_bb.top_left)) {
 				collision_occurring = true;
-				int collided_triangle = mesh_collision_check(second_car, first_car, second_bb.bottom_right);
-				if (collided_triangle != -1) {
-					first_car->collided(collided_triangle);
-					second_car->collided(-1);
-					printf("first car getting hit by second bottom right\n");
-					//first_car->change_color();
+				int victim_triangle = mesh_collision_check(second_car, first_car).victim_index;
+				int attacker_triangle = mesh_collision_check(second_car, first_car).attacker_index;
```

Comment:
can we just call mesh_collision_check once? and get both variables from the same call? it doesn't seem to be a problem now, but it could maybe cause slowdown later on?

Reply:
Sure it just didn't look as nice. I wasn't thinking about efficiency though, I'll do that.